docs/rfcs/042-storage-controller.md

```diff
@@ -0,0 +1,191 @@
+
+## Summary
```

Comment:
A paragraph on what the storage-controller's overall purpose is would be helpful.